src/CodeGen_Hexagon.cpp

```diff
@@ -1517,8 +1517,17 @@ void CodeGen_Hexagon::visit(const Call *op) {
         } else if (op->is_intrinsic(Call::cast_mask)) {
             internal_error << "cast_mask should already have been handled in HexagonOptimize\n";
         }
+    } else if (op->is_intrinsic(Call::bool_to_mask)) {
```

Comment:
This code already exists above in the if vector branch. How about just moving that if outside the branch, and not as an else condition?

At least, the old code should be simplified assuming the if (vector) condition is true, due to the outer if.